contracts/FIFOMarket.sol

```diff
@@ -176,10 +185,11 @@ contract FIFOMarket is
       uint256 removalToAdd = ids[i];
       address supplierAddress = removalToAdd.supplierAddress();
       uint256 removalAmount = batchedAmounts[i];
-      require(
-        _activeSupply[supplierAddress].insertRemovalByVintage(removalToAdd),
-        "Market: Unable to add removal by vintage" // TODO (Gas Optimization): Use custom error
-      );
+      if (
+        !_activeSupply[supplierAddress].insertRemovalByVintage(removalToAdd)
+      ) {
+        revert FailedToAddRemovalToActiveSupply({removalId: removalToAdd});
```

Comment:
rather than returning `bool` from `insertRemovalByVintage` should we just revert if it's not possible?

Reply:
so i'm digging into the implementation of this a bit and `insertRemovalByVintage` returns the return value of `EnumerableSetUpgradable.add()`, which returns 1 if the value was added to the set (that is, if it was not already present) and 0 if it was already present, in which case great, the id is already there (though not great because this should never happen and would indicate a logical problem).

In terms of this error message, I feel like it's misleading because if it's triggered, it's not because we failed to add the removal to active supply, it's because it was already there to begin with.  I think what I'll do is as you suggested, put the revert into `insertRemovalByVintage` and make the error more obvious.